atd-etl/cris_import/create_fact_table_public_views.py

```diff
@@ -0,0 +1,584 @@
+            AND table_schema = 'vz_facts'
+            AND table_name = 'atd_txdot_crashes'
+        order by ordinal_position
+        ), cris as (
+        SELECT
+            'cris' as schema, ordinal_position,
+            column_name, data_type, udt_name, character_maximum_length, numeric_precision,
+            numeric_precision, numeric_scale, is_generated, generation_expression, is_updatable
+        FROM information_schema.columns
+        WHERE true
+            AND table_schema = 'cris_facts'
+            AND table_name = 'atd_txdot_crashes'
+        order by ordinal_position
+        )
+    select vz.column_name as vz_column_name, cris.column_name as cris_column_name,
+        case when vz.column_name is not null then vz.schema
+            when cris.column_name is not null then cris.schema
+        end as schema,
+        case when vz.column_name is not null then vz.column_name
+            when cris.column_name is not null then cris.column_name
+        end as column_name
+    from vz
+        full outer join cris on (vz.column_name = cris.column_name)
+    where true 
+        and (vz.column_name not in ('crash_id') or cris.column_name not in ('crash_id'))
+    order by coalesce(vz.ordinal_position, cris.ordinal_position)
+    """
+    db.execute(sql)
+
+
+    view = """
+    create view public.atd_txdot_crashes as
+        select
+            cris_facts.atd_txdot_crashes.crash_id as crash_id,
+        """
+    columns = []
+    for column in db:
+        print("Column: ", column["vz_column_name"])
+        if column["vz_column_name"] == "longitude_primary":
+            columns.append("ST_X(COALESCE(vz_facts.atd_txdot_crashes.position, cris_facts.atd_txdot_crashes.position)) as longitude_primary")
+        elif column["vz_column_name"] == "latitude_primary":
+            columns.append("ST_Y(COALESCE(vz_facts.atd_txdot_crashes.position, cris_facts.atd_txdot_crashes.position)) as latitude_primary")
+        elif column["vz_column_name"] == "last_update":
+            definition = """
+            case
+                when 
+                    vz_facts.atd_txdot_crashes.last_update is not null
+                    and cris_facts.atd_txdot_crashes.last_update is not null
+                    and vz_facts.atd_txdot_crashes.last_update > cris_facts.atd_txdot_crashes.last_update 
+                        then vz_facts.atd_txdot_crashes.last_update
+                when
+                    vz_facts.atd_txdot_crashes.last_update is not null
+                    and cris_facts.atd_txdot_crashes.last_update is not null
+                    and vz_facts.atd_txdot_crashes.last_update < cris_facts.atd_txdot_crashes.last_update 
+                        then cris_facts.atd_txdot_crashes.last_update
+                else 
+                    coalesce(vz_facts.atd_txdot_crashes.last_update, cris_facts.atd_txdot_crashes.last_update)
+            end as last_update
+            """ 
+        elif column["vz_column_name"] == "last_update":
```

Comment:
Should I expect to see `last_update` in the public crash view? I see that field in both the `cris_facts` and `vz_facts` tables but not in the view. It is in `public.atd_txdot_primaryperson` but not in the person, crash, or unit views.

Reply:
🙈 Oh, thank you Mike! I had totally forgotten to push the `definition` for that column onto the array that later is munged into the VIEW's source. `columns.append(definition)` has been added for crashes. 

But also, while looking at this, it looks like I have duplicated branches to handle `last_update` in this script for 3 of the 4 tables and it's entirely missing from the 4th. I've pushed up a change that addresses these duplications. I /added/ the custom `last_update` handling to the primary person table's special handling. 

Here's [the commit](https://github.com/cityofaustin/atd-vz-data/pull/1288/commits/3bcd68d103737e62182db3f0d196900df60b16d1) that does this, but it doesn't scream off the page exactly what I did. I had two 'elif' blocks for `last_update` and the first one didn't have the `append()` call, so I combined them, and now each main fact table is getting the `last_update` defined correctly and recorded in the view by appending to the `columns` array.

Reply:
Cool, thanks for fixing!